### PR TITLE
Fixes extension fails activation when installing extensions from cli that have a common dependency

### LINKED ISSUE #149309: Extension fails activation when installing extensions from cli that have a common dependency
- VS Code Version: stable and from sources

When launching the server and installing extensions using `--install-extension` that have a common dependency, one of the extensions will fail activating, prompting to reload VS Code.
For example installing, `vscjava.vscode-java-pack` and `pivotal.vscode-spring-boot`
- `vscjava.vscode-java-pack` contains `vscjava.vscode-java-test` which depends on `redhat.java`
- `pivotal.vscode-spring-boot` also depends on `redhat.java`
- `vscjava.vscode-java-pack` installation task finishes first and then trying to activate `vscjava.vscode-java-test` fails as [it doesn't await](https://github.com/microsoft/vscode/blob/93d046904eb3376017e9458105e2511c6122cd35/src/vs/platform/extensionManagement/common/abstractExtensionManagementService.ts#L169) for `redhat.java` installation which is part of `pivotal.vscode-spring-boot` installation task

Steps to Reproduce:

1. Open this branch https://github.com/gitpod-io/spring-petclinic/tree/jp/temp in gitpod or codespaces
2. Observe `vscjava.vscode-java-test` will fail activating and a notification shows prompting to reload vscode 
    error_iamge

cc @sandy081

### PR DESCRIPTION
**Pull request by jeanp413**
This PR fixes #149309

### CODE REVIEW

**Comment by sandy081** [comment]
Will review this in starting of June so that we can get some self hosting time on this.

**Comment by szab100** [comment]
@sandy081 kind reminder 😎

**Comment by sandy081** [comment]
Thanks for the reminder, but I apologise that I could not look this in June.

**Comment by gjsjohnmurray** [comment]
This will also fix https://github.com/microsoft/vscode-remote-release/issues/6020 which @chrmarti opened.

File: src/vs/platform/extensionManagement/common/abstractExtensionManagementService.ts [hunk 1]
```diff
@@ -240,22 +244,43 @@ export abstract class AbstractExtensionManagementService extends Disposable impl
 				}));
 			}
 
+			// Await for extension dependencies that were requested to install beforehand
+			// https://github.com/microsoft/vscode/issues/149309
+			const localDeps = await this.joinAllSettled(alreadyRequestedInstallExtensionTasks.map(task => task.waitUntilTaskIsFinished()));
+			alreadyRequestedInstallExtensionTasks.forEach((task, i) => installResults.push({ local: localDeps[i], identifier: task.identifier, operation: task.operation, source: task.source }));
```
Comment (sandy081) [inline src/vs/platform/extensionManagement/common/abstractExtensionManagementService.ts:250]: It is good that we are waiting until dependencies are installed which fixes one part of the issue but triggering the install event for the same extension multiple times might have side affects. I think the right solution here is to wait until the dependencies are completely installed ie., wait until the install events are triggered for them. I am thinking to come up with such solution.

**Review by sandy081 — CHANGES_REQUESTED** [review]

**Comment by sandy081** [comment]
Closing this in favour of https://github.com/microsoft/vscode/pull/159822

**Pull request closed without merging**